### PR TITLE
BastardKB: Upstream charybdis/3x5

### PR DESCRIPTION
**Pull request by finrod09**
## Description

This change aims to upstream several changes accumulated in the BastardKB fork.

The most extensive of these changes is dropping support for all controllers and shields other than `v2/splinky_3`, which has been flattened into the main keyboard. Older variants have not been sold in kits or prebuilts nor been recommended to self-source for a long time.

The dates GH shows on these commits is misleading as some rewriting of the history was necessary to separate commits by keyboard; The functional changes by @0xcharly have been shipping with kits and prebuilts for months and are well-tested. Changes up to and including https://github.com/qmk/qmk_firmware/commit/3799cd57c363939b583e0a22e2548113b0d559ac are live downstream.

Care has been taken to not lose changes introduced by #23622 and #23627.

Similar changes for all other BastardKB boards are ready, but not yet submitted to reduce the review burden, as they all follow a similar pattern.

Requesting comments.

The full set of related PRs is:
- charybdis/3x5 https://github.com/qmk/qmk_firmware/pull/23638
- charybdis/3x6 https://github.com/qmk/qmk_firmware/pull/23810
- charybdis/4x6 https://github.com/qmk/qmk_firmware/pull/23811
- skeletyl https://github.com/qmk/qmk_firmware/pull/23812
- tbkmini https://github.com/qmk/qmk_firmware/pull/23813
- scylla https://github.com/qmk/qmk_firmware/pull/23814
- dilemma/3x5_2 https://github.com/qmk/qmk_firmware/pull/23815
- dilemma/3x5_3 https://github.com/qmk/qmk_firmware/pull/23817
- dilemma/4x6_4 https://github.com/qmk/qmk_firmware/pull/23816
- tbk (removal) https://github.com/qmk/qmk_firmware/pull/23818

## Types of Changes

- [ ] Core
- [X] Bugfix
- [ ] New feature
- [X] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/bastardkb/charybdis/3x5/rules.mk [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Build Options
+#   change yes to no to disable
+#
+AUDIO_SUPPORTED = no        # Audio is not supported
```
Comment (drashna) [inline keyboards/bastardkb/charybdis/3x5/rules.mk:4]:
```suggestion
```

**Comment by drashna** [comment]
Also, just a heads up, you have a number of keyboard.json files removed.  These replace info.json+rules.mk (eg, you only need the keyboard.json instead of those).  So I/we would prefer if they were not removed, unless there is a specific reason for removing them.

**Comment by finrod09** [comment]
They have been removed as all revisions of the keyboard have been removed entirely - They have been deprecated for a considerable timeframe.
The top-level info.json was amended with the contents of the only revision that remains supported as the legacy-removal was performed downstream on a `master`-based state, then the DD conversion was performed, and info.json was renamed to keyboard.json to fit the `develop` state.

Please let me know if the final result is not to your specifications/expectations.

File: keyboards/bastardkb/charybdis/3x5/rules.mk [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Build Options
+#   change yes to no to disable
+#
+RGB_MATRIX_SUPPORTED = yes  # RGB matrix is supported and enabled by default
+RGBLIGHT_SUPPORTED = yes    # RGB underglow is supported, but not enabled by default
+
+SPLIT_KEYBOARD = yes
+
```
Comment (zvecr) [inline keyboards/bastardkb/charybdis/3x5/rules.mk:8]: This has previously been migrated and should not be reverted.
```suggestion
```

File: keyboards/bastardkb/charybdis/3x5/keyboard.json [hunk 3]
```diff
@@ -1,13 +1,40 @@
...
+        "driver": "ws2812"
         "split_count": [18, 18]
     },
+    "matrix_pins": {
+        "cols": ["GP28", "GP21", "GP6", "GP7", "GP8"],
+        "rows": ["GP26", "GP5", "GP4", "GP9"]
+    },
+    "diode_direction": "ROW2COL",
+    "split": {
+        "soft_serial_pin": "GP1",
```
Comment (zvecr) [inline keyboards/bastardkb/charybdis/3x5/keyboard.json:20]:
```suggestion
        "enabled": true,
        "soft_serial_pin": "GP1",
```

File: keyboards/bastardkb/charybdis/3x5/rules.mk [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Build Options
+#   change yes to no to disable
+#
+RGB_MATRIX_SUPPORTED = yes  # RGB matrix is supported and enabled by default
+RGBLIGHT_SUPPORTED = yes    # RGB underglow is supported, but not enabled by default
```
Comment (zvecr) [inline keyboards/bastardkb/charybdis/3x5/rules.mk:5]: These have been previously removed from the repo, and should not be re-introduced.
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/bastardkb/charybdis/3x5/rules.mk (hunk 2 above)
Comment (finrod09) [reply keyboards/bastardkb/charybdis/3x5/rules.mk]: Thank you for catching the accidental reversals. I guess I was a little eager once I finally had the conflicts resolved, sorry!

**Review by drashna — APPROVED** [review]

**Comment by tzarc** [comment]
As of August 26, 2024, `qmk/qmk_firmware` is no longer accepting VIA-enabled keymaps as these have now transitioned to a repository under the VIA team's control.

As you've submitted a PR containing `via` or VIA-enabled keymap(s), this is your notice that they should be removed from this PR. You should now submit a secondary PR to the [VIA QMK Userspace](https://github.com/the-via/qmk_userspace_via) repository with your associated `via` or VIA-enabled keymaps instead.

**Comment by finrod09** [comment]
The BKB team has decided to stop upstreaming efforts at this time.

**Pull request closed without merging**